File: app/src/main/java/net/osdn/gokigen/blecontrol/lib/ui/fv100/FV100ViewModel.java
package net.osdn.gokigen.blecontrol.lib.ui.fv100;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class FV100ViewModel extends ViewModel
{
    private MutableLiveData<String> mText;

    public FV100ViewModel()
    {
        mText = new MutableLiveData<>();
        mText.setValue(" ");
    }

    void setText(@NonNull String data)
    {
        mText.setValue(data);
    }

    public LiveData<String> getText()
    {
        return (mText);
    }
}

File: app/src/main/java/net/osdn/gokigen/blecontrol/lib/ui/fv100/FV100ViewModel.kt
package net.osdn.gokigen.blecontrol.lib.ui.fv100

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class FV100ViewModel : ViewModel() {
    private val mText: MutableLiveData<String?>

    init {
        mText = MutableLiveData<String?>()
        mText.setValue(" ")
    }

    fun setText(data: String) {
        mText.setValue(data)
    }

    fun addText(data: String) {
        mText.setValue(mText.getValue() + data)
    }

    val text: LiveData<String?>?
        get() = (mText)
}
